micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsConnectionPoolListener.java

```diff
@@ -79,6 +79,11 @@ public void connectionPoolClosed(ConnectionPoolClosedEvent event) {
         waitQueueSize.remove(serverId);
     }
 
+//    @Override
+//    public void connectionPoolCleared(ConnectionPoolClearedEvent event) {
+//        // TODO: Do we need to implement this?
```

Comment:
Is the ConnectionClosedEvent called for each connection when this happens? If so, perhaps we don't need to do anything with this event. The pool is still open and usable after it is cleared, right?

Reply:
Good questions @shakuzen . Seeing this plus your other comment about the other new event types I am in favor of moving these "new event support" to a follow on issue. This ticket is really to move from 3.x -> 4.x and this current proposal puts us in parity w/ 3.x sans "waitqueuesize".

Reply:
It is called when the pool is [invalidated](https://github.com/mongodb/mongo-java-driver/blob/c726328e75252c776857871e9c887a5e4ef12cc5/driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java#L253). This happens in error scenarios but nothing particular happens with the pool in this case. If the connection happens to be closed after this, as @shakuzen said a `ConnectionClosedEvent` will be emitted so we seems to be ok not implementing this.

Reply:
> If the connection happens to be closed

Does it get closed after? I mean, that makes sense that it would, I just did not see it when glancing through the code (I did not pull down locally though either). Either way though, I think its fine to omit it now. If someone wants it they will file an issue.

Reply:
This happens in error scenarios (e.g.: not able to connect) and what happens depends on when do you get the error:

1. It happens before the connection checked out:
This means the connection was not in use, there is a retry but since we din't even get the checked out event, we don't care, eventually there is a timeout and the connection will be closed
1. It happens after the connection checked out: 
This means that the connection was in use, there will be a check in and then a close which is good since it takes care about the events we need

Check out the `ConcurrentPool` (`release`, `prune`, `close`) and the maintenance task in `DefaultConnectionPool` they orchestrate this logic.